src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -0,0 +1,352 @@
+  depthTexture.setStorage(1, Mn::GL::TextureFormat::DepthComponent32F, Size);
+  // Perspective projection
+  {
+    Mn::GL::Framebuffer framebuffer{{{}, Size}};
+    framebuffer
+        .attachTexture(Mn::GL::Framebuffer::ColorAttachment{0}, colorTexture, 0)
+        .attachTexture(Mn::GL::Framebuffer::BufferAttachment::Depth,
+                       depthTexture, 0)
+        .clear(Mn::GL::FramebufferClear::Depth |
+               Mn::GL::FramebufferClear::Color)
+        .bind();
+
+    Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::DepthTest);
+    Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::FaceCulling);
+
+    Mn::Shaders::PhongGL shader{
+        Mn::Shaders::PhongGL::Configuration{}.setLightCount(3)};
+    shader.setProjectionMatrix(perspectiveProj)
+        .setTransformationMatrix(
+            Mn::Matrix4::translation({0.0f, -1.0f, -7.5f}) *
+            Mn::Matrix4::rotationX(-80.0_degf) *
+            Mn::Matrix4::rotationZ(-90.0_degf))
+        .setLightPositions({
+            {10.0f, 10.0f, 10.0f, 0.0f},
+            {-5.0f, -5.0f, 10.0f, 0.0f},
+            {0.0f, 5.0f, -10.0f, 0.0f},
+        })
+        .setLightColors({0xffffff_rgbf, 0xffdddd_rgbf, 0xddddff_rgbf})
+        .draw(mesh);
+
+    MAGNUM_VERIFY_NO_GL_ERROR();
+    CORRADE_COMPARE_WITH(
+        framebuffer.read({{}, Size}, {Mn::PixelFormat::RGBA8Unorm}),
+        Cr::Utility::Path::join(TEST_ASSETS,
+                                "hbao_tests/van-gogh-room.color.png"),
+        (Mn::DebugTools::CompareImageToFile{}));
+    CORRADE_COMPARE_WITH(
+        framebuffer.read({{}, Size}, {Mn::PixelFormat::Depth32F}),
+        Cr::Utility::Path::join(TEST_ASSETS,
+                                "hbao_tests/van-gogh-room.depth.exr"),
+        (Mn::DebugTools::CompareImageToFile{}));
+  }
+  // Ortho projection
+
+  setOrthoProjection(Magnum::Vector2(4.0f, 4.0f));
+  {
+    Mn::GL::Framebuffer framebuffer{{{}, Size}};
+    framebuffer
+        .attachTexture(Mn::GL::Framebuffer::ColorAttachment{0}, colorTexture, 0)
+        .attachTexture(Mn::GL::Framebuffer::BufferAttachment::Depth,
+                       depthTexture, 0)
+        .clear(Mn::GL::FramebufferClear::Depth |
+               Mn::GL::FramebufferClear::Color)
+        .bind();
+
+    Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::DepthTest);
+    Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::FaceCulling);
+
+    Mn::Shaders::PhongGL shader{
+        Mn::Shaders::PhongGL::Configuration{}.setLightCount(3)};
+    shader.setProjectionMatrix(orthoProj)
+        .setTransformationMatrix(
+            Mn::Matrix4::translation({0.0f, -1.0f, -7.5f}) *
+            Mn::Matrix4::rotationX(-80.0_degf) *
+            Mn::Matrix4::rotationZ(-90.0_degf))
+        .setLightPositions({
+            {10.0f, 10.0f, 10.0f, 0.0f},
+            {-5.0f, -5.0f, 10.0f, 0.0f},
+            {0.0f, 5.0f, -10.0f, 0.0f},
+        })
+        .setLightColors({0xffffff_rgbf, 0xffdddd_rgbf, 0xddddff_rgbf})
```

Comment:
Given all the code duplication here, this test would maybe be better instanced, similarly as the other one? With this as input:

```cpp
const struct {
    const char* name;
    const char* filenameSuffix;
    Mn::Matrix4 projection;
} GenerateTestDataData[]{
    {"perspective", "", 
        Mn::Matrix4::perspectiveProjection(45.0_degf, 1.0f, 0.1f, 100.0f)},
    {"orthographic", "-ortho",
        Mn::Matrix4::orthographicProjection(size, 0.1f, 100.0f)},
};
```